opentimelineio_contrib/adapters/advanced_authoring_format.py

```diff
@@ -814,6 +811,31 @@ def _contains_something_valuable(thing):
     return True
 
 
+def attach_markers(timeline):
+    '''
+    Search for markers and attach them to their corresponding track
+    '''
+    markers = []
+
+    # Get all markers in the given unsimplified timeline. Markers come in as non
+    # audio or video tracks
+    for child in list(timeline.each_child()):
```

Comment:
This would be better as:
`for track in each_child(descended_from_type=otio.schema.Track):`
then you'll only need to check for markers.